content/post/2017-05-22-using-asdf-vm.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Using ASDF-vm
+subtitle: Solve All of Your Version Problems
+date: 2017-05-21
+author: Connor Rigby
+draft: false
+tags: ["nerves"]
+---
+
+Recently there has been a number of breaking changes in the Erlang/Elixir world that have caused Nerves-based projects to fail to compile, causing a lot of confusion in slack. Follow along as we solve all your version inconsistencies with ASDF-vm.
+
+<!--more-->
+
+# What caused the problem?
+Erlang released OTP 20 into bleeding edge repositories last week, and in response to that Elixir released `v1.4.4`. If you follow the basic rules of [semver](http://semver.org/) you might spot a big red flag here.
+
+* OTP did a MAJOR version bump
+* Elixir (which is heavily dependent on OTP) did a PATCH
+
+Granted OTP 20 is still RC, there is still a versioning inconsistency that ironically came from Elixir's `version` module.
+So heres a short description of Elixir and OTP versions.
+
+* If you want Elixir 1.4.4 you _*NEED*_ OTP 20.
```

Comment:
Is there a bug report on the Elixir project that we can refer to on this? I think people will be skeptical, since Elixir 1.4.4 really looks like it runs on OTP 19.